Fix thumbnails not showing on < API 21

tossj opened the pull request:
Preview images for streams, VODs, games, etc. fail to load on devices running an Android version before Lollipop. This is due to the implementation of a custom transformation for the Glide library.

Glide's `BitmapTransformation` instances should not recycle `toTransform` in transform method or it will cause a `java.lang.IllegalStateException: Cannot pool recycled bitmap.`, and prevent thumbnails from appearing.

See [`BitmapTransformation.transform`](https://bumptech.github.io/glide/javadocs/440/com/bumptech/glide/load/resource/bitmap/BitmapTransformation.html#transform-com.bumptech.glide.load.engine.bitmap_recycle.BitmapPool-android.graphics.Bitmap-int-int-) for more info:
> The provided Bitmap, toTransform, should not be recycled or returned to the pool. Glide will automatically recycle and/or reuse toTransform if the transformation returns a different Bitmap.

### Known issues
- A very small part of a thumbnail's top corners can be white sometimes. This white is the area outside of the rounded rectangle that was painted with the bitmap. Pulling to refresh causes these white corners to not appear. The white corners are not large enough to interfere with seeing the thumbnail, and the advantages of getting the thumbnail to appear at all outweigh this minor imperfection.

tossj commented:
@Perflyst Could you review this PR when you get a chance?

samfundev commented:
Thanks for the PR. I was able to fix the white corners by changing the XML and I got [this](https://user-images.githubusercontent.com/6759716/88716243-c8ec2580-d0ec-11ea-99b0-6abd5e0d0ccc.png). The problem is that the bottom corners aren't rounded. I'm thinking that it might just be better to not rounded the cards on pre-Lollipop devices.

tossj commented:
> Thanks for the PR. I was able to fix the white corners by changing the XML and I got [this](https://user-images.githubusercontent.com/6759716/88716243-c8ec2580-d0ec-11ea-99b0-6abd5e0d0ccc.png). The problem is that the bottom corners aren't rounded. I'm thinking that it might just be better to not rounded the cards on pre-Lollipop devices.

Another thing I tried was just removing the condition to round the top corners of the images. This resulted in the bottom corners being rounded still but the top corners were of the image, and weren't rounded. It looked nearly identical to having the rounded image corners, and because an extra Glide transformation isn't being appied, it saves computation as well. I imagine doing this way would also avoid the need for pre-Lollipop specific logic in the XML?

However, I think there is a way to make this current PR work as is because when I pull to refresh on the games page the white corners go away and the top corners only appear round. I was thinking either the CardView or the layout in the CardView needs to be refreshed somehow, but I'm not entirely sure.

samfundev commented:
Well if you're able to find a way to get it to work then I'll pull this in but otherwise I think it might just be best to remove the transformation that applies the rounding.

tossj commented:
@samfundev I've fixed the white corners on the preview images. They were caused by the animation code drawing a rectangle with the background color before drawing in the image. The left over background rectangle was what showing through the transparent corners of the rounded preview image. I've replaced the animation code with Glide's own fade in.

I've also cleaned up some of the preview loading code to better use Glide's features. For example, the `.placeholder` method can handle when the loading errs, and when the `previewURL` doesn't exist, so I've removed the `if` branches that were checking these. Glide also maintains its own cache, so I've removed the code to reuse the bitmap from the `PreviewTarget`. I would have also removed the `PreviewTarget` itself and loaded directly into the image view, but `ChannelsAdapter` uses the bitmap from the `PreviewTarget` for something, so I left it in.

Please review when you get a chance.

samfundev commented:
Sorry for the wait, I've just got around to reviewing this.

First, the Glide transitions weren't actually being used. This is because they wasn't being called in the `onResourceReady` method. I also happened to spot that we need to use cross fade when transitioning. Because Twitch avatars can have transparency which without a cross fade would leave the placeholder behind the avatar. And we're no longer using `setPicassoShowImageAnimationTwo` so it can be removed. These are all pretty easy fixes so I went ahead and fixed them up.

Thanks for the PR! I'll rebase and merge this in.
